fs/any_version_test.go

```diff
@@ -0,0 +1,82 @@
+package fs
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/hc-install/product"
+)
+
+func TestAnyVersionValidate(t *testing.T) {
+	t.Parallel()
+
+	productWithInvalidBinaryName := product.Terraform
+	productWithInvalidBinaryName.BinaryName = func() string { return "invalid!" }
+
+	testCases := map[string]struct {
+		av          AnyVersion
+		expectedErr error
+	}{
+		"no-fields-set": {
+			av:          AnyVersion{},
+			expectedErr: fmt.Errorf("must use either ExactBinPath or Product + ExtraPaths"),
+		},
+		"ExactBinPath-and-Product": {
+			av: AnyVersion{
+				ExactBinPath: "/test",
+				Product:      &product.Terraform,
+			},
+			expectedErr: fmt.Errorf("use either ExactBinPath or Product + ExtraPaths, not both"),
+		},
+		"ExactBinPath-empty": {
+			av: AnyVersion{
+				ExactBinPath: "",
+			},
+			expectedErr: fmt.Errorf("must use either ExactBinPath or Product + ExtraPaths"),
+		},
+		"ExactBinPath-relative": {
+			av: AnyVersion{
+				ExactBinPath: "test",
+			},
+			expectedErr: fmt.Errorf("expected ExactBinPath (\"test\") to be an absolute path"),
+		},
+		"ExactBinPath-absolute": {
+			av: AnyVersion{
+				ExactBinPath: "/test",
+			},
+		},
+		"Product-incorrect-binary-name": {
+			av: AnyVersion{
+				Product: &productWithInvalidBinaryName,
```

Comment:
Nitpick: I think this would be fine as in-line declared struct with in-lined `BinaryName` function.

Reply:
Sure thing! Will adjust in a second.